libraries/ms-common/inc/interrupt.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+// Shared interface elements across all devices for interrupts. Each device will also have internal
+// device specific functionality which exist in their respective interrupt_mcu.h files (do not
+// include that file directly instead reference it through a module such as gpio_it.h).
+
+#include <stdbool.h>
+
+// A generic interrupt callback with context.
+typedef void (*interrupt_callback)(void *context);
```

Comment:
Done. I just pass it as a const char * of __FUNC__ from the calling function. Does that make sense to you? It seemed the easiest method.